### PR TITLE
[RFC] vim-patch:8.0.1633: a TextChanged autocmd triggers when it is defined

### PR DESCRIPTION
**Pull request by blueyed**
Problem:    A TextChanged autocmd triggers when it is defined after creating a
            buffer.
Solution:   Set b_last_changedtick when opening a buffer. (Hirohito Highlight,
            closes vim/vim#2742)
https://github.com/vim/vim/commit/8c64a36e40b8746404f7151abe6849393396af10

TODO:
- [ ] requires 8.0.0365?! (see https://github.com/neovim/neovim/pull/8474#issuecomment-397953628)

### CODE REVIEW

**Comment by blueyed** [comment]
The bug itself (https://github.com/vim/vim/pull/2742) is present in v0.3.0, but not v0.2.2.
(It causes Neomake to lint files on startup / when loading them, through `TextChanged`.)

**Comment by janlazo** [comment]
Feel free to include any vim patches required for this PR. I will work on vim 8.0.0252 and other non-terminal patches before 8.0.0693 in my backlog.

File: src/nvim/buffer.c [hunk 1]
```diff
@@ -295,6 +295,13 @@ open_buffer (
   }
   save_file_ff(curbuf);                 // keep this fileformat
 
+  /* Set last_changedtick to avoid triggering a TextChanged autocommand right
+   * after it was added. */
+  curbuf->b_last_changedtick = CHANGEDTICK(curbuf);
+#ifdef FEAT_INS_EXPAND
```
Comment (Shougo) [inline src/nvim/buffer.c]: The ifdef is not needed for neovim.

**Comment by justinmk** [comment]
@blueyed since https://github.com/neovim/neovim/pull/8618 was merged I think this needs a rebase? Is there anything else blocking this?

File: src/nvim/buffer.c [hunk 2]
```diff
@@ -295,6 +295,13 @@ open_buffer (
   }
   save_file_ff(curbuf);                 // keep this fileformat
 
+  /* Set last_changedtick to avoid triggering a TextChanged autocommand right
+   * after it was added. */
+  curbuf->b_last_changedtick = CHANGEDTICK(curbuf);
```
Comment (janlazo) [inline src/nvim/buffer.c]: https://github.com/neovim/neovim/commit/e1c6109e62b4e1d12ebdab67140f82cc7dc2c0d1#diff-aea5ac6a6784947c12e7550683f5938dR1202

CHANGEDTICK(curbuf) ==> buf_get_changedtick(curbuf)

**Comment by blueyed** [comment]
Feel free to pick this up, please - I will not have much time for this currently.

**Pull request closed without merging**